speaker/layer2_controller.go

```diff
@@ -93,13 +91,10 @@ func (c *layer2Controller) ShouldAnnounce(l log.Logger, name string, toAnnounce
 		return "notOwner"
 	}
 
-	// we select the nodes with at least one matching l2 advertisement
-	forPool := c.speakersForPool(l, name, pool, nodes)
-	var availableNodes []string
+	speakerMap := c.speakersForPool(l, name, pool, nodes)
+	availableNodes := convertMapToSlice(speakerMap)
```

Comment:
the previous name of the function made the logic clearer, I'd stick to that

Reply:
I do not see why `usableNodes` which just transforms the map to array looks better, but done

Reply:
the old version was nodesWithActiveSpeakers, so it's clear what we are picking

Reply:
correctly my bad